### PR TITLE
Implement App SDK 2.0

### PR DESCRIPTION
**Pull request by lkostrowski**
- App SDK 2.0
- Migrate domain -> saleorApiUrl
- Fix Macaw UI theme blinking
- verifyJWT for client calls

### CODE REVIEW

File: src/modules/trpc/procedure-with-graphql-client.ts [hunk 1]
```diff
@@ -21,6 +22,12 @@ const attachAppToken = middleware(async ({ ctx, next }) => {
     });
   }
 
+  try {
+    await verifyJWT({ appId: authData.appId, token: ctx.token, apiUrl: saleorApiUrl });
```
Comment (krzysztofwolski) [inline src/modules/trpc/procedure-with-graphql-client.ts]:
```suggestion
    await verifyJWT({ appId: authData.appId, token: ctx.token, apiUrl: saleorApiUrl });
```

Does this part works? Can't see `saleorApiUrl` defined in the scope

Comment (lkostrowski) [reply src/modules/trpc/procedure-with-graphql-client.ts]: it doesnt, I need to wait for new app-sdk to provide it

File: src/modules/trpc/procedure-with-graphql-client.ts [hunk 2]
```diff
@@ -12,6 +13,13 @@ const attachAppToken = middleware(async ({ ctx, next }) => {
     });
   }
 
+  if (!ctx.token) {
+    throw new TRPCError({
+      code: "BAD_REQUEST",
+      message: "Missing domain in request",
```
Comment (krzysztofwolski) [inline src/modules/trpc/procedure-with-graphql-client.ts]:
```suggestion
      message: "Missing token in request",
```

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

**Review by krzysztofwolski — APPROVED** [review]

File: src/pages/api/register.ts [hunk 3]
```diff
@@ -2,8 +2,27 @@ import { createAppRegisterHandler } from "@saleor/app-sdk/handlers/next";
 
 import { saleorApp } from "../../../saleor-app";
 
+const allowedUrlsPattern = process.env.ALLOWED_DOMAIN_PATTERN;
```
Comment (krzysztofwolski) [inline src/pages/api/register.ts:5]: Would consider moving `allowedUrlsPattern` decalaration/reading from env into `saler-app.ts`.
Not hard requirements, but rather my prefererence is to keep most of reading configuration from the ENV in the one file.

**Review by krzysztofwolski — APPROVED** [review]
Looking good, one suggestion added.

File: src/pages/api/register.ts (hunk 3 above)
Comment (lkostrowski) [reply src/pages/api/register.ts:5]: Good, agree. Lets think about this after release. Technically it will be easy to do it in SDK.

**Pull request merged**